apps/openssl.c

```diff
@@ -243,7 +222,6 @@ int main(int argc, char *argv[])
     arg.size = 0;
 
     /* Set up some of the environment. */
-    default_config_file = make_config_name();
     bio_in = dup_bio_in(FORMAT_TEXT);
     bio_out = dup_bio_out(FORMAT_TEXT);
     bio_err = dup_bio_err(FORMAT_TEXT);
```

Comment:
also unclear if the return value of dup_bio_xxx should be checked likewise.

Reply:
or do what app_malloc does and fail/exit into dup_bio routine.

Reply:
Note there are already calls in apps/lib/apps.c
where the return value is checked.

Reply:
Something like this, and remove the one :) place it's checked:
```
diff --git a/apps/lib/apps.c b/apps/lib/apps.c
index dbc9be344d..38ff5adfe3 100644
--- a/apps/lib/apps.c
+++ b/apps/lib/apps.c
@@ -2363,6 +2363,17 @@ int raw_write_stdout(const void *buf, int siz)
 }
 #endif
 
+static BIO *bio_check(BIO *bp, const char *what, int format)
+{
+    if (bp == NULL) {
+        BIO_printf(bio_err, "%s: Could not duplicate %s (mode 0x%x)\n",
+                opt_getprog(), what, format);
+        ERR_print_errors(bio_err);
+        exit(1);
+    }
+    return bp;
+}
+
 /*
  * Centralized handling of input and output files with format specification
  * The format is meant to show what the input and output is supposed to be,
@@ -2372,14 +2383,15 @@ int raw_write_stdout(const void *buf, int siz)
  */
 BIO *dup_bio_in(int format)
 {
-    return BIO_new_fp(stdin,
-                      BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0));
+    int mode = BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0); 
+
+    return bio_check(BIO_new_fp(stdin, mode), "stdin", mode);
 }
 
 BIO *dup_bio_out(int format)
 {
-    BIO *b = BIO_new_fp(stdout,
-                        BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0));
+    int mode = BIO_NOCLOSE | (FMT_istext(format) ? BIO_FP_TEXT : 0);
+    BIO *b = BIO_new_fp(stdout, mode);
     void *prefix = NULL;
 
 #ifdef OPENSSL_SYS_VMS
@@ -2393,7 +2405,7 @@ BIO *dup_bio_out(int format)
         BIO_set_prefix(b, prefix);
     }
 
-    return b;
+    return bio_check(b, "stdout", mode);
 }
 
 BIO *dup_bio_err(int format)
```

Fixing ```dup_bio_err``` is left as an exercise for the reader :)

Reply:
@richsalz This would render the errorhandling in app_get_pass pointless:
```C
        } else if (strcmp(arg, "stdin") == 0) {
            pwdbio = dup_bio_in(FORMAT_TEXT);
            if (pwdbio == NULL) {
                BIO_printf(bio_err, "Can't open BIO for stdin\n");
                return NULL;
            }
```

Reply:
to keep it simple, I would just prefer
```C
if (bio_in == NULL || bio_out == NULL || bio_err == NULL)
  app_bail_out(...);
```

Reply:
yes it would make that pointless, shrug.  it would also fix engine.c, two cases in each of s_client an s_server. And then look at the callers of bio_open_default.  Seems worth it to me.

Reply:
good point.

Reply:
@levitte if this still cherry-picks to 1.1.1 I'd approve it now,
and let that for another PR.

Reply:
Other PR sounds good to me

Reply:
If I do it, will it get reviewed?  Or are you going to do it.

Reply:
Please do it.  I will review.

Reply:
Oh do feel free!